### ISSUE TITLE
Make chessboard setup more efficient

### ISSUE THREAD

**Issue by Levi-Lesches** [Learning-Python-Team/Chess#16]
Right now each square is initialized to an empty piece in `Board.__init__`. Then, in `Board.setup`, each square is looped over and potentially replaced with another piece. 

We can probably do that all in one go, and halve the work we need to do for setup. I'm thinking about moving the "which piece goes on this square?" logic to `Square.__init__`, which already has access to its row & column

**Comment by ZeroOne010101** [comment]
wouldnt that make init more messy? also, we can use the current setup for a 'play again' prompt after a winning condition is fullfilled

**Comment by Levi-Lesches** [comment]
Delegating the logic to `Square.reset` and calling it in `Square.__init__` instead of blindly assigning a `NonePiece`, we can solve both problems! If we do it right, we can simply go: 
```python
# in Board
def reset(self): 
  for square in self: 
    square.reset()
```

And `Square.reset` will contain the logic for deciding which piece to place.

**Comment by ZeroOne010101** [comment]
i see. i like it, but how do you plan to assign default values to the squares?

**Comment by Levi-Lesches** [comment]
What do you mean default?

**Comment by ZeroOne010101** [comment]
you call reset on square, right? how does square know what it should reset to?

**Comment by Levi-Lesches** [comment]
Yes. Because `Square` has `row` and `column` properties!

**Comment by ZeroOne010101** [comment]
so you give square a self.default with the setup logic?

**Comment by Levi-Lesches** [comment]
Oh, I see what you're saying. So basically cache the right piece after `Square.reset` figures it out, right? That way `Square.reset` can say later `if self.default is not None: self.piece = self.default`.

**Comment by ZeroOne010101** [comment]
yeah. otherwise that'd be setup but more complicated

**Comment by Levi-Lesches** [comment]
Ok. I'll get to work on it and open a PR.

**Comment by Levi-Lesches** [comment]
Closed with #17
